src/components/fdk-fieldset/styled.ts

```diff
@@ -12,12 +12,19 @@ const Fieldset = styled.fieldset`
   }
 `;
 
-const Legend = styled.legend`
+const Legend = styled.legend<{ boxed?: boolean }>`
   display: flex;
   flex-direction: column;
   width: 100%;
-  padding: 15px;
-  background: ${({ theme }) => theme.fdk.colors.neutrals.skyblue};
+  padding: 0;
+
+  ${({ boxed = true }) =>
+    boxed &&
+    css`
+      background: ${({ theme }) => theme.fdk.colors.neutrals.skyblue};
+      padding: 15px;
+    `}
```

Comment:
It might be better to switch the condition on its head, i.e. boxed prop to default to false. That way, you do not need a default case (boxed = true) and your components will look cleaner by opting in to certain props (no boxed={false} syntax).